File: Desafios em C#/1049 - Animal.cs
using System; 

//Developed by: @LucasMarcuzo
class URI1049 {

    static void Main(string[] args) { 

        string a, b, c, total="";

        a=Convert.ToString(Console.ReadLine());
        b=Convert.ToString(Console.ReadLine());
        c=Convert.ToString(Console.ReadLine());
        

        if(a == "vertebrado"){

            if(b == "ave"){

                if(c == "carnivoro"){

                    total = "aguia";

                }else if(c == "onivoro"){

                    total = "pomba";

                }

            }else if(b == "mamifero"){

                if(c == "onivoro"){

                    total = "homem";

                }else if(c == "herbivoro"){

                    total = "vaca";

                }
            }
        
        
        }else if(a == "invertebrado"){

            if(b == "inseto"){

                if(c == "hematofago"){

                    total = "pulga";

                }else if(c == "herbivoro"){

                    total = "lagarta";

                }

            }else if(b == "anelideo"){

                if(c == "hematofago"){

                    total = "sanguessuga";

                }else if(c == "onivoro"){

                    total = "minhoca";

                }
            }
        }

            Console.WriteLine(total);

    }

}

File: Desafios em JavaScript/1049 - Animal.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var a, b, c, total;

a=String(lines.shift());
b=String(lines.shift());
c=String(lines.shift());

if(a === "vertebrado")
{
    if(b === "ave")
    {
        if(c === "carnivoro")
        {
            total = String("aguia");
        }
        else if(c === "onivoro")
        {
            total = String("pomba");
        }
    }
    else if(b === "mamifero")
    {
        if(c === "onivoro")
        {
            total = String("homem");
        }
        else if(c === "herbivoro")
        {
            total = String("vaca");
        }
    }      
}
else if(a === "invertebrado")
{
    if(b === "inseto")
    {
        if(c === "hematofago")
        {
            total = String("pulga");
        }
        else if(c === "herbivoro")
        {
            total = String("lagarta");
        }
    }
    else if(b === "anelideo")
    {
        if(c === "hematofago")
        {
            total = String("sanguessuga");
        }
        else if(c === "onivoro")
        {
            total = String("minhoca");
        }
    }
}

console.log(total);